src/main/java/uk/gov/justice/laa/crime/cfecrime/utils/FullMeansTestOutcomeCalculator.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.justice.laa.crime.cfecrime.utils;
+
+import lombok.extern.slf4j.Slf4j;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.FullAssessmentResult;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MagCourtOutcome;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.CaseType;
+import uk.gov.justice.laa.crime.cfecrime.cma.enums.MeansTestOutcome;
+import java.util.HashSet;
+import java.util.Set;
+
+@Slf4j
+public class FullMeansTestOutcomeCalculator {
+
+    public static MeansTestOutcome getFullMeansTestOutcome(FullAssessmentResult result, CaseType caseType, MagCourtOutcome magCourtOutcome){
+        log.debug("Get the outcome of the Full Means Test result = {} caseType = {} magCourtOutcome = {}", result, caseType, magCourtOutcome);
+
+        //Fail data
+        Set<CaseType> caseTypesHeardInMagistratesCourt = new HashSet<CaseType>();
+        caseTypesHeardInMagistratesCourt .add(CaseType.COMMITAL);
+        caseTypesHeardInMagistratesCourt .add(CaseType.SUMMARY_ONLY);
+        caseTypesHeardInMagistratesCourt .add(CaseType.EITHER_WAY);
+
+        MeansTestOutcome meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
+        if (result != null && caseType != null ) {
+
+            if (result == FullAssessmentResult.PASS) {
+                    //All Eligible with no contribution
+                    meansTestOutcome = MeansTestOutcome.ELIGIBLE_WITH_NO_CONTRIBUTION;
+            }
+            if (result == FullAssessmentResult.FAIL) {
```

Comment:
This section feels backwards - if we fail, I would expect the default to be ineligible with some special cases? Maybe we should have a 3-way if statement instead of a default to make this clearer?

Reply:
changed and poked with @starswan